robotoff/app/api.py

```diff
@@ -166,26 +166,39 @@ def on_get(self, req: falcon.Request, resp: falcon.Response):
 
 class RandomInsightResource:
     def on_get(self, req: falcon.Request, resp: falcon.Response):
+        response: JSONType = {}
+        count: int = req.get_param_as_int("count", min_value=1, default=25)
+        page: int = req.get_param_as_int("page", min_value=1, default=1)
+        
         insight_type: Optional[str] = req.get_param("type")
         country: Optional[str] = req.get_param("country")
         value_tag: Optional[str] = req.get_param("value_tag")
         server_domain: Optional[str] = req.get_param("server_domain")
         count: int = req.get_param_as_int("count", default=1, min_value=1, max_value=50)
 
         keep_types = [insight_type] if insight_type else None
-        insights: List[ProductInsight] = list(
-            get_insights(
-                keep_types=keep_types,
-                country=country,
-                value_tag=value_tag,
-                order_by="random",
-                server_domain=server_domain,
-                limit=count,
-            )
+        get_insights_ = functools.partial(
+            get_insights,
+            keep_types=keep_types,
+            country=country,
+            server_domain=server_domain,
+            value_tag=value_tag,
+            order_by="random",
+            server_domain=server_domain,
         )
 
-        resp.media = {"insights": [insight.serialize() for insight in insights]}
+        offset: int = (page - 1) * count
+        insights = [i.serialize() for i in get_insights_(limit=count, offset=offset)]
```

Comment:
Is there any value to add a page to a random query ? The random query will gives you different results every time, so pagination does not really have sense (but if we cached the results).

Maybe we can just ignore them (result is different every time), because taking them into account gonna make things less performant for no gain.

Reply:
Yes, for random it does not make sens